pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1427,12 +1427,18 @@ func (r *ReconcileClusterDeployment) addClusterDeploymentFinalizer(cd *hivev1.Cl
 func (r *ReconcileClusterDeployment) removeClusterDeploymentFinalizer(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
 
 	cd = cd.DeepCopy()
+
+	// DeprovisionUnderwaySeconds metric must be cleared prior to deletion of the deprovision finalizer.
+	// This is due to controller-runtime caching behavior that may cause a cd to appear
+	// to have a deprovision finalizer after the finalizer has been removed, due to a stale
+	// cd object in the cache.
+	clearDeprovisionUnderwaySecondsMetric(cd, cdLog)
+
 	controllerutils.DeleteFinalizer(cd, hivev1.FinalizerDeprovision)
 	if err := r.Update(context.TODO(), cd); err != nil {
 		return err
 	}
 
-	clearDeprovisionUnderwaySecondsMetric(cd, cdLog)
 	// Clear ClusterSyncFailing metric
 	hivemetrics.ClearClusterSyncFailingSecondsMetric(cd.Namespace, cd.Name, cdLog)
```

Comment:
I thought we said not to move it, but I can't remember the reasoning. To me, it makes sense to move it as well.

Reply:
It's the one below we said not to move, because we would double-count this CD as deleted if we hit the cache bug. And that one isn't subject to clearing/deletion or coordination with other controllers, so it's "safe" to leave where it is.

This one, clearing the clustersync failing metric, is of the same ilk as the deprov underway, so it _should_ be moved.